src/networkcloud/azext_networkcloud/operations/common_ssh.py

```diff
@@ -0,0 +1,144 @@
+        """generates ssh keys in user .ssh folder"""
+
+        key_name = "id_rsa"
+        private_key_path = os.path.join(os.path.expanduser("~"), ".ssh", key_name)
+        public_key_path = os.path.join(
+            os.path.expanduser("~"), ".ssh", f"{key_name}.pub"
+        )
+        logger.warning(
+            "Attempted to find or generate SSH key files %s and %s.pub under ~/.ssh to allow SSH access "
+            "to the vm. If using machines without permanent storage, back up "
+            "your keys to a safe location.",
+            key_name,
+            key_name,
+        )
+        return [{"keyData": keys.generate_ssh_keys(private_key_path, public_key_path)}]
+
+    @staticmethod
+    def get_ssh_keys_from_path(values):
+        """get ssh keys from the provided path"""
+
+        key_list = []
+        for path in values:
+            # Casting is necessary to get away from the custom type AAZSimpleType wrapping the string we want
+            path = str(path)
+            if os.path.isdir(path):
+                content = ""
+                for key in os.listdir(path):
+                    if key.endswith(".pub"):
+                        with open(
+                            os.path.join(
+                                path,
+                                key,
+                            ),
+                            "r",
+                            encoding="utf-8",
+                        ) as k:
+                            content = k.read()
+                        try:
+                            if keys.is_valid_ssh_rsa_public_key(content):
+                                key_list.append({"keyData": content})
+                        except Exception as exception:
+                            raise InvalidArgumentValueError(
+                                f"Unsupported Key {key} is provided.\nContent:\n{content}."
+                            ) from exception
+                if len(key_list) == 0:
+                    raise InvalidArgumentValueError(
+                        f"No public keys found in the path: {path}"
+                    )
+            elif os.path.isfile(path):
+                with open(path, "r", encoding="utf-8") as k:
+                    content = k.read()
+                try:
+                    if keys.is_valid_ssh_rsa_public_key(content):
+                        key_list.append({"keyData": content})
+                except Exception as excep:
+                    raise InvalidArgumentValueError(
+                        f"Unsupported Key {path} is provided.\nContent:\n{content}"
+                    ) from excep
+            else:
+                raise InvalidArgumentValueError(
+                    f"An invalid directory or key was provided: {path}"
+                )
+        return key_list
+
+    @staticmethod
+    def add_ssh_key_action(values):
+        """validate and add ssh key to the list"""
+
+        key_data = []
+        for key in values:
```

Comment:
If the `values` comes from to_serialized_data(), the key is string type by default.

Reply:
As the calling function could not be updated to use to_serialized_data() due to an error (details in previous comment), leaving this as this. please let us know if we missed anything.